merkle/hashers/tree_hasher.go

```diff
@@ -42,3 +48,47 @@ type MapHasher interface {
 	// TODO(gbelvin): Replace Size() with BitLength().
 	Size() int
 }
+
+var (
+	maxHashers = trillian.HashStrategy(len(trillian.HashStrategy_value))
```

Comment:
There's a dangerous assumption here that `len(trillian.HashStrategy_value)` is greater than the highest proto tag used. This works for now, but will break if we retire a value from the enum or if we simply skip tag numbers.